### ISSUE TITLE
Ball pivoting mesh reconstruction doesn't consider point/vertex normals

### ISSUE THREAD

**Issue by clausmichele** [cnr-isti-vclab/meshlab#420]
When using Ball Pivoting to reconstruct a mesh from a Point Cloud, the algorithm recomputes the normals, leading to a wrong direction in some parts of the mesh (see image). Recomputing face normals or inverting them does not solve the problem.
Since I have to apply a texture to the generated mesh, this generates artifacts (see image).
The point cloud has correctly oriented normals, computed setting the point of view.

Point cloud with correctly oriented normals:
![immagine](https://user-images.githubusercontent.com/31700619/52843923-ba31ce80-3103-11e9-9096-7b9635a4a816.png)

Generated mesh with Ball Pivoting with applied texture:
![immagine](https://user-images.githubusercontent.com/31700619/52843845-940c2e80-3103-11e9-97ce-93f3fb59c94d.png)

**Comment by Hugal31** [comment]
Reproduced on v2016.12

**Comment by mcallieri** [comment]
yes, ball pivoting only uses coordinates. The original algorithm was developed to triangulate sparse, non-uniform clouds of points without normals, and that is the version we have in MeshLab, at the moment.
It could be nice to take in account also normals (maybe controlled with a parameter), but I have no idea when it will be possible to add this.

**Comment by Hugal31** [comment]
That is strange, because the description of the algorithm states "Given a point cloud **with** normals it reconstructs a surface using the Ball Pivoting Algorithm."
